Content.Server/WhiteDream/BloodCult/BloodRites/BloodRitesSystem.cs

```diff
@@ -234,4 +238,36 @@ Entity<BloodstreamComponent> target
         rites.Comp.StoredBlood -= bloodCost;
         return true;
     }
+
+    private void ConsumePuddles(EntityUid origin, Entity<BloodRitesAuraComponent> rites)
+    {
+        var coords = Transform(origin).Coordinates;
+
+        var lookup = _lookup.GetEntitiesInRange<PuddleComponent>(
+            coords,
+            rites.Comp.PuddleConsumeRadius,
+            LookupFlags.Uncontained);
+
+        foreach (var puddle in lookup)
+        {
+            if (!TryComp(puddle, out SolutionContainerManagerComponent? solutionContainer))
+                return;
```

Comment:
```suggestion
                continue;
```
Otherwise it will exit early the first time it encounters a puddle without a solution, which while it shouldn't happen, would mean it immediately stops checking all the rest of the puddles.